### ISSUE TITLE
CAST Object to Object with Nullable subcolumns doesn't work

### ISSUE THREAD

**Issue by Avogar** [ClickHouse/ClickHouse#36558]
I expect it should work:
```
:) select CAST(CAST('{"x" : 1}', 'Object(\'json\')'), 'Object(Nullable(\'json\'))')

SELECT CAST(CAST('{"x" : 1}', 'Object(\'json\')'), 'Object(Nullable(\'json\'))')

Query id: fb057b45-b72a-4d22-9b38-a3136cdf2465


0 rows in set. Elapsed: 0.002 sec.

Received exception from server (version 22.5.1):
Code: 53. DB::Exception: Received from localhost:9000. DB::Exception: Cast to Object can be performed only from flatten named Tuple, Map or String. Got: Object('json'): While processing CAST(CAST('{"x" : 1}', 'Object(\'json\')'), 'Object(Nullable(\'json\'))'). (TYPE_MISMATCH)
```

@CurtizJ what do you think?

**Comment by CurtizJ** [comment]
Need to fix it. It's quite easy, BTW.

**Comment by CurtizJ** [comment]
Fixed in #36564.
